python/ml4ir/base/model/scoring/scoring_model.py

```diff
@@ -303,7 +303,11 @@ def __update_metrics(self, inputs, y_true, y_pred):
         - Currently only support pre-compiled Keras metrics
         """
         # Compute metrics on primary label
-        self.compiled_metrics.update_state(y_true, y_pred)
+        try:
+            mask = inputs[self.feature_config.get_mask("node_name")]
+        except:
```

Comment:
@mohazahran Only catch the error you expect. Catch all will lead to silent failures hard to trace. I assume you mean a `KeyError` here?

Also, can you remove the code below which does the same thing again to fetch the mask again?

Reply:
done